lib/voicebase/v3/client.rb

```diff
@@ -0,0 +1,153 @@
+module VoiceBase
+  module V3
+    module Client
+      def self.extended(client, args = {})
+        client.api_host     = client.args[:host] || ENV.fetch('VOICEBASE_V3_API_HOST', 'https://apis.voicebase.com')
+        client.api_endpoint = client.args[:api_endpoint] || ENV.fetch('VOICEBASE_V3_API_ENDPOINT', '/v3')
+      end
+
+      def authenticate!
+        auth = {:username => @auth_key, :password => @auth_secret}
+        response = VoiceBase::Response.new(
+          self.class.get(
+            uri + '/access/users/admin/tokens',
+            basic_auth: auth,
+            headers: {
+              'User-Agent'   => @user_agent,
+              'Accept'       => 'application/json'
+            }
+          ), api_version)
+        @token = VoiceBase::Client::Token.new(response.tokens.any? && response.tokens.first.fetch("token"))
+      rescue NoMethodError => ex
+        raise VoiceBase::AuthenticationError, response.status_message
+      end
+
+      def upload_media(args = {}, headers = {})
+        require_media_file_or_url!(args)
+        r = ::VoiceBase::Request::MultipartBuilder.new(headers: default_headers(headers))
+
+        if args[:config]
+          r.add_attachment(VoiceBase::Request::HashPart.new(name: "configuration", hash: args[:config]))
+        end
+
+        if args[:media_url]
+          r.add_attachment(VoiceBase::Request::TextPart.new(name: "mediaUrl", body: args[:media_url]))
+        end
+
+        if args[:media_file]
+          r.add_attachment(VoiceBase::Request::FilePart.new(name: "media", file: args[:media_file]))
+        end
+
+        #TODO: make metadata an object
+        if args[:metadata]
+          r.add_attachment(VoiceBase::Request::HashPart.new(name: "metadata", hash: args[:metadata]))
+        end
+
+        response = self.class.post(
+            uri + '/media',
+            headers: r.headers,
+            body: r.body
+        )
+        VoiceBase::Response.new(response, api_version)
+      end
+
+      # I presume this method exists for parity with the V1 API however we are not using it
+      def get_media(args = {}, headers = {})
+        raise ArgumentError, "Missing argument :media_id" unless args[:media_id]
+        url = if args[:media_id]
```

Comment:
why is there an `if` statement here given it was just checked above?

Reply:
I also didn't touch this. this is from v2. https://github.com/usertesting/voicebase-client-ruby/blob/master/lib/voicebase/v2/client.rb#L43-L59

Reply:
maybe the new commit would make more sense https://github.com/usertesting/voicebase-client-ruby/pull/16/commits/28bfe248708eef26141fc76c659031438f35e8dc